src/config.rs

```diff
@@ -338,6 +363,35 @@ impl SetOpt for Proxy<http::Uri> {
     }
 }
 
+/// No Proxy configuration.
+///
+/// The default configuration is for the system proxy to be used.
+///
+/// See [`HttpClientBuilder::noproxy`](crate::HttpClientBuilder::noproxy)
+/// for configuring a client's no proxy list.
+#[derive(Clone, Debug)]
+pub enum NoProxy {
+    /// Disable proxy usage entirely.
+    All,
```

Comment:
Is there a difference between setting `CURLOPT_PROXY` to `""` and setting `CURLOPT_NOPROXY` to `"*"`? If the end result is the same then it would probably be better to keep these as separate things, like this:

```rust
// Change from http::Uri to impl Into<Option<http::Uri>>, which is effectively compatible.
// Setting to `None` explicitly disables proxy.
pub fn proxy(mut self, proxy: impl Into<Option<http::Uri>>) -> Self; 

// More narrow focus, disable for a specific list of hosts.
pub fn proxy_blacklist(mut self, hosts: impl IntoIterator<Item = String>) -> Self;
```